log4j-layout-template-json/src/main/java/org/apache/logging/log4j/layout/template/json/resolver/MessageResolver.java

```diff
@@ -134,7 +134,10 @@ private static void resolveString(
             jsonWriter.writeObjectStart();
             jsonWriter.writeObjectKey(fallbackKey);
         }
-        if (message instanceof StringBuilderFormattable) {
+        if (message instanceof CharSequence) {
```

Comment:
Is this an optimization for SimpleMessage? I wonder if there's something we can do better in `jsonWriter.writeString(StringBuilderFormattable)`? in the SimpleMessage/ReusableSimpleMessage case, I suspect we would do better using `getFormattedMessage()` as `StringBuilder.append(String)` is faster than iterating over a charsequence. Note that this would allocate in the case that the message value is a non-string charsequence.

StringBuilderFormattable handling writes from the original message to an intermediate StringBuilder, which is escaped into the final StringBuilder. I think we could probably optimize this process, for instance I think we always call `quoteString(formattableBuffer, 0, length);` which uses `StringBuilder.append(cs,start,end)` rather than `StringBuilder.append(cs)` -- unfortunately only the latter appears to be special-cased for AbstractStringBuilder.

Reply:
You have a point @carterkozak. This said, I think we better address this in a future iteration to get this PR rolling.